### PR TITLE
Newsletter settings: re-organise

### PR DESCRIPTION
**Pull request by simison**
## Proposed Changes

* Re-organise Newsletter settings page, add new section copies

image

Note that copy is changed in https://github.com/Automattic/wp-calypso/pull/83059 which needs rebasing if this PR merges first.

## Testing Instructions

* Go to Settings → Newsletter and confirm everything works across Simple & Atomic sites.
* You can disable Subscriptions module for Jetpack site from _Discussion settings_. Then look at Newsletter settings page and you should see a banner, and be able to enable module via the banner CTA.

    Screenshot 2023-10-16 at 20 15 57

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by simison** [comment]
A related bug that needs sorting out: https://github.com/Automattic/wp-calypso/issues/82815

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9304984

Thank you @simison for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9304984).

File: client/my-sites/site-settings/settings-newsletter/newsletter-categories-section/newsletter-categories-section.tsx [hunk 1]
```diff
@@ -1,50 +1,31 @@
...
-			{ /* @ts-expect-error SettingsSectionHeader is not typed and is causing errors */ }
-			<SettingsSectionHeader
-				id="newsletter-categories-settings"
-				title={ translate( 'Newsletter categories settings' ) }
-				showButton
-				onButtonClick={ handleSubmitForm }
-				disabled={ disabled }
-				isSaving={ isSavingSettings }
-			/>
-
```
Comment (simison) [inline client/my-sites/site-settings/settings-newsletter/newsletter-categories-section/newsletter-categories-section.tsx:47]: Moved settings header from this file to `main.jsx` for consistency with other sections which also now have their header in the main file.

**Comment by simison** [comment]
Still needs category tests addressed after changes. They feel a tad bit extensive for a simple component... I think I'd simply reduce it a bit.

File: client/my-sites/site-settings/settings-newsletter/main.tsx [hunk 2]
```diff
@@ -104,36 +120,99 @@ const NewsletterSettingsForm = wrapSettingsForm( getFormSettings )( ( {
...
+			/>
+			<Card className="site-settings__card">
+				<EmailsTextSetting
 					disabled={ disabled }
-					isSavingSettings={ isSavingSettings }
-					savedSubscriptionOptions={ savedSubscriptionOptions }
 					updateFields={ updateFields }
+					value={ subscription_options }
 				/>
-			</form>
```
Comment (edanzer) [inline client/my-sites/site-settings/settings-newsletter/main.tsx:121]: I see we're moving from two form elements in the old code to one in the new code. I assume that's fine (and it tests fine), but I was wondering why those were originally separated?

**Comment by edanzer** [comment]
This is testing as expected in Simple Sites. 

And the code changes look straightforward, even if there's a lot moving around here. 

Notes:
 - As already noted, there are some failing unit tests.
 - I added one question about moving from two forms to one.
 - I tested on atomic, including turning off/on the subscribe module. It was mostly fine. But there was a bit of weirdness on my atomic site unrelated to this PR (I was able to duplicate the same weirdness on production). Like my calypso instance would periodically stop loading or show unrelated errors. Not sure what's going on there - I think my atomic site is borked. But again, anything that was not working was also not working on prod. If you'd like more assurance, I can set up and test with a new WoA site tomorrow. For my part, given the nature of changes, and everything else testing well, I'd be fine deploying. But I'd be fine doing another round of testing in the morning too.

File: client/my-sites/site-settings/settings-newsletter/main.tsx (hunk 2 above)
Comment (simison) [reply client/my-sites/site-settings/settings-newsletter/main.tsx:121]: I think it was just the side-effect how development — The Categories bit was quite well isolated from the rest of the settings on this page, which is fair.

I don't think each section needs its own form, and we can simplify by submitting all changes on the page from all "Save" buttons instead.

Similarly I moved (https://github.com/Automattic/wp-calypso/pull/82871#discussion_r1361013820) the header component out of categories to the main component — undoing a bit of the clear boundary/isolation but I think it makes files overall easier to parse and more consistent.

**Pull request merged**